REGRESSION (iOS 16): Characters outside a font-face's unicode-range attribute don't render when characters in that range are present

litherum opened the pull request:
#### 7b14522b6997c35323528528352225edfbf1d935
<pre>
REGRESSION (iOS 16): Characters outside a font-face&apos;s unicode-range attribute don&apos;t render when characters in that range are present
<a href="https://bugs.webkit.org/show_bug.cgi?id=241831">https://bugs.webkit.org/show_bug.cgi?id=241831</a>
&lt;rdar://95646820&gt;

Reviewed by Alan Bujtas.

When we fall off the end of the font fallback list, we create a &quot;system fallback&quot; font. It&apos;s possible for
this font to be a web font (see <a href="https://github.com/w3c/csswg-drafts/issues/7449">https://github.com/w3c/csswg-drafts/issues/7449</a> for discussion about this).
If it is a web font, we need to preserve the CreationData of the font so it can be transferred across IPC.

Test: fast/text/font-face-fall-off-end.html

* LayoutTests/fast/text/font-face-fall-off-end-expected.html:
* LayoutTests/fast/text/font-face-fall-off-end.html:
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::FontCache::systemFallbackForCharacters):

Canonical link: <a href="https://commits.webkit.org/252096@main">https://commits.webkit.org/252096@main</a>
</pre>

alanbaradlay commented on Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
```diff
@@ -1565,7 +1565,10 @@ RefPtr<Font> FontCache::systemFallbackForCharacters(const FontDescription& descr
 
-    FontPlatformData alternateFont(substituteFont, platformData.size(), syntheticBold, syntheticOblique, platformData.orientation(), platformData.widthVariant(), platformData.textRenderingMode());
+    const FontPlatformData::CreationData* creationData = nullptr;
+    if (const auto& existingCreationData = originalFontData.platformData().creationData())
```
could you reuse the local platformData here?

litherum committed:
REGRESSION (iOS 16): Characters outside a font-face's unicode-range attribute don't render when characters in that range are present
https://bugs.webkit.org/show_bug.cgi?id=241831
<rdar://95646820>

Reviewed by Alan Bujtas.

When we fall off the end of the font fallback list, we create a "system fallback" font. It's possible for
this font to be a web font (see https://github.com/w3c/csswg-drafts/issues/7449 for discussion about this).
If it is a web font, we need to preserve the CreationData of the font so it can be transferred across IPC.

Test: fast/text/font-face-fall-off-end.html

* LayoutTests/fast/text/font-face-fall-off-end-expected.html:
* LayoutTests/fast/text/font-face-fall-off-end.html:
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::FontCache::systemFallbackForCharacters):

Canonical link: https://commits.webkit.org/252096@main
